### ISSUE TITLE
`setWholeStringToRGBColor.py` example doesn't work with firmware version 3.51

### ISSUE THREAD

**Issue by gregsadetsky** [zranger1/pixelblaze-client#21]
### Environment

- OS: macOS 12.7
- Python: 3.12.2
- Pixelblaze: 3.51
- pixelblaze-client: 1.1.3

### Minimal reproducible example

I tried running the [setWholeStringToRGBColor.py](https://github.com/zranger1/pixelblaze-client/blob/main/examples/setWholeStringToRGBColor.py) example but I got `Sorry, I don't know how to compile with firmware version 3.51.` as the error

### Expected Behavior

I expected to be able to set the strip to a single color

### Actual Behavior

I get the error message

**Comment by zranger1** [comment]
Thank you for reporting this!  I'll get it fixed shortly.

**Comment by gregsadetsky** [comment]
hey don't mean to rush you, but we're very excited about our pixelblazer controlled LED strip -- if you ever have a moment to fix this it would be really appreciated. thank you! :-)

**Comment by zranger1** [comment]
What is your use case for this?  Perhaps I can suggest an easier way to get there.  If it's just changing pixel colors, it is much, much faster to run a simple pattern with exported color variables on the Pixelblaze, and use "SetActiveVariables" to change them from Python. 

This whole _"download the compiler from the pixelblaze firmware, hack it to run in a browser emulation, then manually package and upload the compiled byte code"_, is a mechanism I honestly didn't think anybody (including me) was ever going to use unless writing an offline IDE for Pixelblaze or something like that.  In addition to being super sensistive to firmware changes, it's also very slow compared to the alternatives.

Wizard changed the underlying mechanism when he added the leader/follower system, so I have some reverse engineering to  do to figure out how the protocol changed.  I've got the compiler working again, but at the moment, I'm super slammed with IRL work and burning man prep, so it may be a while before I can work out the new upload system.

**Comment by gregsadetsky** [comment]
ah ok ok ! thanks a lot -- I did end up going with that -- your pattern from https://github.com/zranger1/pixelblaze-client/blob/main/examples/animate.py#L29 i.e. `export var pixels = array(pixelCount); ...`, and then just sending the whole pixel string as a variable via websocket (i.e. via set active variables)

I originally thought that there would have been a more direct way of sending the entire sequence of pixel rgb values, but in the end, I guess that it's really all the same i.e. I send pixels to my own script that immediately renders it

all good here then :-) thanks for helping me with this, and for all of these sample scripts!!

**Comment by zwily** [comment]
Oh wow, I didn't realize that that's how that worked... I thought the code got compiled on the device, but I guess it makes sense that the more powerful browser would do it... That's a bummer. I wanted to use your library to create a script that would make it easier for me to combine code - I have a bunch of global code that needs to go into every pattern, and it's a hassle to manage it manually. But I guess that this is kind of a non-starter...

**Comment by zranger1** [comment]
Yeah, you download the current compiler from the Pixelblaze, strip it of a bunch of interactive web goo, and run it on a minimalist browser emulator.   Of course, it's totally possible to make a stand-alone PB compiler, and I've occasionally been tempted. 

But I respect Ben's work on the Pixelblaze, and I want him to succeed wildly with it.  So I try to choose things that will help him rather than make it easy for his competitors.  (Even doing this remote compile trick in Python is walking a little close to the line vs. the "I will not decompile or reverse engineer" in the Pixelblaze EULA.)

Anyway, my crew is out building our art car at Burning Man now, and we'll freeze the software probably Friday, so I should be able to get back to this next week.

**Comment by zranger1** [comment]
Just pushed the fix for this (updated version is 1.1.4) to the repo and to pypi.

**Comment by gregsadetsky** [comment]
thank you so much!
